x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/trigger/schedule/engine/TickerScheduleTriggerEngine.java

```diff
@@ -81,17 +85,22 @@ public synchronized void start(Collection<Watch> jobs) {
 
     @Override
     public void stop() {
+        logger.debug("Stopping engine");
+        isRunning.set(false);
         schedules.clear();
         ticker.close();
     }
 
     @Override
-    public synchronized void pauseExecution() {
+    public void pauseExecution() {
+        logger.debug("Pausing engine");
+        isRunning.set(false);
         schedules.clear();
     }
 
     @Override
     public void add(Watch watch) {
+        logger.debug("Adding watch [{}] to engine (engine is running: {})", watch.id(), isRunning.get());
```

Comment:
might want to make this a trace level message since this could result in thousands of log messages.